### PR TITLE
inc-715: rust consumer can stop processing messages at specific timestamp

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

File: rust_snuba/src/strategies/processor.rs [hunk 1]
```diff
@@ -34,7 +35,18 @@ pub fn make_rust_processor(
         partition: Partition,
         offset: u64,
         timestamp: DateTime<Utc>,
+        stop_at_timestamp: Option<i64>,
     ) -> anyhow::Result<Message<BytesInsertBatch<RowData>>> {
+        // Don't process any more messages
+        if let Some(stop) = stop_at_timestamp {
+            if stop < timestamp.timestamp() {
+                let payload = BytesInsertBatch::default();
```
Comment (fpacifici) [inline rust_snuba/src/strategies/processor.rs:43]: I assume this is an empty batch.

File: rust_snuba/src/strategies/processor.rs [hunk 2]
```diff
@@ -214,9 +244,21 @@ impl<TResult: Clone, TNext: Clone> MessageProcessor<TResult, TNext> {
             timestamp: msg.timestamp,
         };
 
+        if let Some(stop_timestamp) = self.stop_at_timestamp {
+            if metadata.timestamp.timestamp() > stop_timestamp {
+                anyhow::bail!("message is after stop timestamp");
+            }
+        }
+
```
Comment (fpacifici) [inline rust_snuba/src/strategies/processor.rs:252]: I don't think I understand why this is needed. 
Isn't the condition already in`result_to_next_msg`

**Comment by ayirr7** [comment]
Is it possible to have a unit test that checks that any of the processors returns an empty batch for a message that exceeds stop timestamp?

File: rust_snuba/src/strategies/processor.rs (hunk 2 above)
Comment (ayirr7) [reply rust_snuba/src/strategies/processor.rs:252]: +1

**Review by fpacifici — APPROVED** [review]

File: rust_snuba/src/strategies/processor.rs (hunk 2 above)
Comment (lynnagara) [reply rust_snuba/src/strategies/processor.rs]: sorry, this was an old iteration and i forgot to push, updated

**Review by ayirr7 — APPROVED** [review]

**Pull request merged**